### PR TITLE
bugfix: rollback fail when insert

### PR DESCRIPTION
**Pull request by 0000005**
### Ⅰ. Describe what this PR did
1. add configuration `doNotCareOnUpdateTimestampWhenInsert`, the default value of this param is `true`.
2. Building after image of insert SQL will ignore the column that decorated by `OnUpdateTimestamp` If `doNotCareOnUpdateTimestampWhenInsert` is true.

### Ⅱ. Does this pull request fix one issue?

This pr fixed this issue(#3036). Other useful information is shown on this issue too.

### CODE REVIEW

**Comment by a364176773** [comment]
[ERROR] Failures: 
[ERROR]   SelectForUpdateExecutorTest.testDoExecute:96 Unexpected exception thrown: io.seata.common.exception.ShouldNeverHappenException: [xid:xid]get tablemeta failed
[ERROR] Errors: 
[ERROR]   DeleteExecutorTest.testAfterImage:101 » ShouldNeverHappen [xid:null]get tablem...
[ERROR]   DeleteExecutorTest.testBeforeImage:90 » ShouldNeverHappen [xid:null]get tablem...
[ERROR]   MultiExecutorTest.testBeforeImageAndAfterImages:100 » ShouldNeverHappen [xid:n...
[ERROR]   UpdateExecutorTest.testAfterImage:102 » ShouldNeverHappen [xid:xid]get tableme...
[ERROR]   UpdateExecutorTest.testBeforeImage:91 » ShouldNeverHappen [xid:xid]get tableme...
[ERROR]   TableRecordsTest.testBuildRecords:107 » ShouldNeverHappen [xid:null]get tablem...
[ERROR]   TableRecordsTest.testPkRow:88 » ShouldNeverHappen [xid:null]get tablemeta fail...
[ERROR]   MysqlTableMetaCacheTest.getTableMetaTest_0:87 » ShouldNeverHappen [xid:null]ge...
[ERROR]   MysqlTableMetaCacheTest.refreshTest_0:136 » ShouldNeverHappen [xid:null]get ta...

**Comment by 81519434** [comment]
Sorry, please check again. @a364176773

**Comment by jsbxyyx** [comment]
@81519434 the pr function use for other database is there a problem?

**Comment by 81519434** [comment]
I tested the PR on MySQL, it seems good. 
Because I don't have any other database on my computer, so other databases are not tested.
But the only thing that related the database is to determine whether the column is "on update timestamp".
```java
          while (Objects.nonNull(vsColumns) && vsColumns.next()) {
                String columnName = vsColumns.getString("COLUMN_NAME");
                int dataType = vsColumns.getInt("DATA_TYPE");
                ColumnMeta col = tm.getAllColumns().get(columnName);
                if (col != null && dataType == Types.TIMESTAMP) {
                    col.setIsOnUpdateTimestamp("Yes");
                }
            }
```
I think that piece of code should be fine when running in other databases.
Can you help me to verify the question?
@jsbxyyx

**Comment by 81519434** [comment]
If you confirm that the pr is not working in other databases, I will find a way to fix it.

**Comment by 81519434** [comment]
The test case is very simple. You can find out in my issue(#3036).

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java [hunk 1]
```diff
@@ -405,6 +422,14 @@ protected TableRecords buildTableRecords(Map<String, List<Object>> pkValuesMap)
         }
     }
 
+    private boolean canIgnoreOnUpdateTimestamp(TableMeta tableMeta, String columnName) {
+        if (DO_NOT_CARE_ON_UPDATE_TIMESTAMP_WHEN_INSERT) {
+            ColumnMeta currentColumn = tableMeta.getColumnMeta(ColumnUtils.delEscape(columnName, getDbType()));
+            return "Yes".equals(currentColumn.getIsOnUpdateTimestamp());
```
Comment (jsbxyyx) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java:428]: `Yes` extract constant.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java [hunk 2]
```diff
@@ -138,6 +141,14 @@ private TableMeta resultSetMetaToSchema(ResultSetMetaData rsmd, DatabaseMetaData
 
                 tm.getAllColumns().put(col.getColumnName(), col);
             }
+            while (Objects.nonNull(vsColumns) && vsColumns.next()) {
+                String columnName = vsColumns.getString("COLUMN_NAME");
+                int dataType = vsColumns.getInt("DATA_TYPE");
+                ColumnMeta col = tm.getAllColumns().get(columnName);
+                if (col != null && dataType == Types.TIMESTAMP) {
+                    col.setIsOnUpdateTimestamp("Yes");
```
Comment (jsbxyyx) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/MysqlTableMetaCache.java:149]: same up.

File: common/src/main/java/io/seata/common/DefaultValues.java [hunk 3]
```diff
@@ -54,6 +54,8 @@
     boolean DEFAULT_TRANSACTION_UNDO_DATA_VALIDATION = true;
     String DEFAULT_TRANSACTION_UNDO_LOG_SERIALIZATION = "jackson";
     boolean DEFAULT_ONLY_CARE_UPDATE_COLUMNS = true;
+    boolean DEFAULT_DO_NOT_CARE_ON_UPDATE_TIMESTAMP_WHEN_INSERT = true;
```
Comment (l81893521) [inline common/src/main/java/io/seata/common/DefaultValues.java:57]: I think the default value change to false will be better when set up a new configuration.

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

**Pull request closed without merging**